modules/core/02-client/keeper/client.go

```diff
@@ -24,18 +25,12 @@ func (k Keeper) CreateClient(
 	}
 
 	clientID := k.GenerateClientIdentifier(ctx, clientState.ClientType())
+	clientStore := k.ClientStore(ctx, clientID)
 
-	k.Logger(ctx).Info("client created at height", "client-id", clientID, "height", clientState.GetLatestHeight().String())
-
-	// verifies initial consensus state against client state and initializes client store with any client-specific metadata
-	// e.g. set ProcessedTime in Tendermint clients
-	if err := clientState.Initialize(ctx, k.cdc, k.ClientStore(ctx, clientID), consensusState); err != nil {
+	if err := clientState.Initialize(ctx, k.cdc, clientStore, consensusState); err != nil {
```

Comment:
It would be good to extend the existing test by making sure we can query the client state for the given clientID (consensus state we can skip query for since it may not be set)

Reply:
Done! I think we could really do with cleaning up the 02-client test cases to use the testing suite properly.

Reply:
oh yea! You get a glimpse into how the repo looked back in the day! It's one of the main places that hasn't been cleaned up